tests/unit_tests/storage/test_subject_manager.py

```
@@ -0,0 +1,504 @@
+        sub_agent_name = "error_sub_agent"
+        _add_sub_agent_with_namespace(real_agent_config, sub_agent_name, "test_ns")
+        updater = SubjectUpdater(real_agent_config)
+
+        _seed_metric(updater.metrics_storage)
+
+        # Sub-agent storage doesn't have the metric seeded, so update_entry will raise ValueError.
+        # SubjectUpdater should catch it gracefully (lines 67-68).
+        updater.update_metrics_detail(SUBJECT_PATH, METRIC_NAME, {"description": "should not crash"})
+
+        # Verify main storage was still updated despite sub-agent failure
+        results = updater.metrics_storage.search_all_metrics(subject_path=SUBJECT_PATH + [METRIC_NAME])
```

Ruff (0.15.2)

[warning] Consider `[*SUBJECT_PATH, METRIC_NAME]` instead of concatenation

Replace with `[*SUBJECT_PATH, METRIC_NAME]`

(RUF005)